### PR TITLE
Binding formatter to UI

### PR DESCRIPTION
**Pull request by moosq**

### CODE REVIEW

File: internal/lsp/configuration.go [hunk 1]
```diff
@@ -22,3 +23,31 @@ func (s *Server) Config(section string) interface{} {
 	}
 	return v
 }
+
+var regFormatCapability = protocol.RegistrationParams{
+	Registrations: []protocol.Registration{
+		{ID: "FormCap",
```
Comment (5nord) [inline internal/lsp/configuration.go:29]: "FormCap" seems arbitrary and ambiguous, how about something like "ttcn3.formatting" instead?

File: internal/lsp/formatter.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+func (s *Server) formatting(ctx context.Context, params *protocol.DocumentFormattingParams) ([]protocol.TextEdit, error) {
+	b, err := fs.Content(params.TextDocument.URI.SpanURI().Filename())
+	if err != nil {
+		log.Debug("formatting: ", err.Error())
+		return nil, nil
+	}
+	var out bytes.Buffer
+	p := printer.NewCanonicalPrinter(&out)
+	if params.Options.InsertSpaces {
+
```
Comment (5nord) [inline internal/lsp/formatter.go:24]:
```suggestion
```

File: internal/lsp/formatter.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		p.Indent = fmt.Sprintf("%*s", params.Options.TabSize, " ")
+	}
+
+	if err := p.Fprint(b); err != nil {
+		log.Debug("formatting: ", err.Error())
```
Comment (5nord) [inline internal/lsp/formatter.go:29]:
```suggestion
		log.Debug("formatting:", err.Error())
```

File: internal/lsp/server.go [hunk 4]
```diff
@@ -64,6 +64,10 @@ func (s serverState) String() string {
 	return fmt.Sprintf("(unknown state: %d)", int(s))
 }
 
+type ServerConfig struct {
```
Comment (5nord) [inline internal/lsp/server.go:67]:
```suggestion
type Config struct {
```

File: internal/lsp/server.go [hunk 5]
```diff
@@ -84,7 +88,8 @@ type Server struct {
 	diagsMu sync.Mutex
 	diags   map[string][]protocol.Diagnostic
 
-	testCtrl *TestController
+	testCtrl     *TestController
+	serverConfig ServerConfig
```
Comment (5nord) [inline internal/lsp/server.go:92]: I find `Server.serverConfig` is too repetetive. How about:
```suggestion
	Config
```

File: internal/lsp/server.go [hunk 6]
```diff
@@ -64,6 +64,10 @@ func (s serverState) String() string {
 	return fmt.Sprintf("(unknown state: %d)", int(s))
 }
 
+type ServerConfig struct {
+	UseOwnFormatter bool
```
Comment (5nord) [inline internal/lsp/server.go:68]: I find the adjective too specific. How about:
```suggestion
	UseFormatter bool
```
or 
```suggestion
	FormatTTCN3 bool
```

File: internal/lsp/configuration.go [hunk 7]
```diff
@@ -22,3 +23,31 @@ func (s *Server) Config(section string) interface{} {
...
+
+		if format == s.serverConfig.UseOwnFormatter {
+			return nil
+		}
+		if s.serverConfig.UseOwnFormatter {
+			log.Debug("didChangeConfiguration: unregister formatter capability\n")
+			s.client.UnregisterCapability(ctx, &unregFormatCapability)
+		} else {
+			log.Debug("didChangeConfiguration: register formatter capability\n")
+			s.client.RegisterCapability(ctx, &regFormatCapability)
```
Comment (5nord) [inline internal/lsp/configuration.go:48]: I think (un)regFormatCapability does not need to be a global variable. We probably can inline it?

**Review by 5nord — COMMENTED** [review]
Before we merge this PR, we should find a better configuration name for `ttcn3.server.UseOwnFormatter` first.

File: internal/lsp/server.go (hunk 5 above)
Comment (moosq) [reply internal/lsp/server.go]: `Config`as a field name clashes with `Config`method

**Pull request merged**